### PR TITLE
Terminate block production loop when shutting down witness plugin #1314

### PR DESCRIPTION
**Pull request by cogutvalera**
Terminate block production loop when shutting down witness plugin #1314

handle enumeration value  in switch (block_production_loop)

### CODE REVIEW

**Comment by abitmore** [comment]
I don't think this is necessary. If want the `ilog`, we can just add it in the first place.

**Comment by abitmore** [comment]
Oh, this is to fix the `switch` warning. Then why there is no `default`?

**Comment by cogutvalera** [comment]
Thanks ! Going to add `default` now.

File: libraries/plugins/witness/witness.cpp [hunk 1]
```diff
@@ -235,6 +241,12 @@ block_production_condition::block_production_condition_enum witness_plugin::bloc
       case block_production_condition::exception_producing_block:
          elog( "exception producing block" );
          break;
+      case block_production_condition::shutdown:
+         ilog( "shutdown producing block" );
+         return result;
+      default:
+         elog( "unknown condition while producing block" );
```
Comment (pmconrad) [inline libraries/plugins/witness/witness.cpp]: Please add the actual ``result`` to the log message.

Comment (cogutvalera) [reply libraries/plugins/witness/witness.cpp]: ok sure ! Thanks ! 
Do you mean `default` case only ? Or both cases with `shutdown` also ?

Thanks !

**Comment by pmconrad** [comment]
The default clause has the downside that it will prevent a similar future issue from being detected at compile-time. It is therefore not always a good idea to add one.
In this specific case, logging an error is ok.

File: libraries/plugins/witness/witness.cpp (hunk 1 above)
Comment (pmconrad) [reply libraries/plugins/witness/witness.cpp]: Only the default case. In the other cases the message is sufficient to identify the case.

**Comment by cogutvalera** [comment]
@pmconrad I'm absolutely agree with you that `default` case won't prevent a similar issue at compile-time except this specific case with error log, absolutely agree, that's why I've added error log.

**Pull request merged**